contracts/SupportedAssets.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+
+/**
+ * @title SupportedAssets
+ * @dev Contract storing list of available assets for investing
+ */
+contract SupportedAssets is Ownable {
+
+
+  mapping(bytes32 => address) assetToAddress;
+  bytes32[] assetKeys;
+
+
+  /* ========== SET FUNCTIONS ========== */
+
+
+  function setAsset(bytes32 _asset, address _address) external onlyOwner {
+    require(_asset != "", "Cannot set an empty string asset.");
+    require(assetToAddress[_asset] == address(0), "Cannot set an asset that has already been set.");
+
+    assetKeys.push(_asset);
+    assetToAddress[_asset] = _address;
+
+    emit AssetAdded(_asset);
```

Comment:
WE could add address to the event

Reply:
now included because the array of `Asset` struct is returned